File: packages/typia/src/schemas/metadata/MetadataEscaped.ts
import { IMetadataSchema } from "@typia/interface";

import { ClassProperties } from "../../typings/ClassProperties";
import { IMetadataDictionary } from "./IMetadataDictionary";
import { MetadataSchema } from "./MetadataSchema";

export class MetadataEscaped {
  public readonly original: MetadataSchema;
  public readonly returns: MetadataSchema;

  /** @ignore */
  private constructor(props: ClassProperties<MetadataEscaped>) {
    this.original = props.original;
    this.returns = props.returns;
  }

  /** @internal */
  public static from(
    props: IMetadataSchema.IEscaped,
    dict: IMetadataDictionary,
  ): MetadataEscaped {
    return MetadataEscaped.create({
      original: MetadataSchema.from(props.original, dict),
      returns: MetadataSchema.from(props.returns, dict),
    });
  }

  /** @internal */
  public static create(
    props: ClassProperties<MetadataEscaped>,
  ): MetadataEscaped {
    return new MetadataEscaped(props);
  }

  public getName(): string {
    return this.returns.getName();
  }

  public toJSON(): IMetadataSchema.IEscaped {
    return {
      original: this.original.toJSON(),
      returns: this.returns.toJSON(),
    };
  }
}

File: packages/typia/native/core/schemas/metadata/MetadataEscaped.go
package metadata

type IMetadataSchema_IEscaped struct {
	Original *IMetadataSchema
	Returns  *IMetadataSchema
}

type MetadataEscaped struct {
	Original *MetadataSchema
	Returns  *MetadataSchema
}

func MetadataEscaped_from(props IMetadataSchema_IEscaped, dict IMetadataDictionary) *MetadataEscaped {
	return MetadataEscaped_create(MetadataEscaped{
		Original: MetadataSchema_from(props.Original, dict),
		Returns:  MetadataSchema_from(props.Returns, dict),
	})
}

func MetadataEscaped_create(props MetadataEscaped) *MetadataEscaped {
	return &MetadataEscaped{
		Original: props.Original,
		Returns:  props.Returns,
	}
}

func (obj *MetadataEscaped) GetName() string {
	return obj.Returns.GetName()
}

func (obj *MetadataEscaped) ToJSON() IMetadataSchema_IEscaped {
	return IMetadataSchema_IEscaped{
		Original: obj.Original.ToJSON(),
		Returns:  obj.Returns.ToJSON(),
	}
}
